store/root.js

```diff
@@ -137,23 +142,15 @@ class Root extends Store {
       return
     }
 
-    // Compare strings and numbers
-    // since we do not really know what type of ID the API exposes
-    // eslint-disable-next-line eqeqeq
-    const hasTargetId = (dataProvider) => id == dataProvider.id
+    const dataProviderInt = parseInt(id, 10)
 
     // Probably a repeated request. No need to change
-    if (this.dataProvider != null && hasTargetId(this.dataProvider)) return
+    // Compare strings and numbers
+    // since we do not really know what type of ID the API exposes
+    if (this.dataProvider?.id === dataProviderInt) return
 
-    // Check access rights
-    // TODO: Should be moved or ideally removed
-    const dataProvider = this.dataProviders.find(hasTargetId)
-    if (dataProvider == null) {
-      const dpStr = `DataProvider#${id}`
-      throw new AccessError(`${this.user} does not have access to the ${dpStr}`)
-    }
+    this.dataProvider = this.user.getDataProviderById(dataProviderInt)
```

Comment:
We probably should start fetching data providers directly from the API, not from the user. Let's think about on the next step.